File: Lista11_Python/Ex03_PegadaDeCarbono/carro.py
from interfacePegadaDeCarbono import *

class Carro(PegadaDeCarbono):

    def __init__(self, marca, modelo, combustivel):

        if (combustivel != "alcool" and combustivel != "gasolina" and combustivel != "hibrido"):
            print("Combustível deve ser: alcool ou gasolina")
            return

        self.marca = marca
        self.modelo = modelo
        self.combustivel = combustivel
        self.quilometragem = 0

    def __str__(self):
        string = "CARRO\n"
        string += "Marca: "+self.marca+"\nModelo: "+self.modelo+"\nTipo de combustivel: "+self.combustivel
        return string

    def getMarca(self):
        return self.marca

    def move(self, distancia):
        self.quilometragem += distancia

    def getPegadaCarbono(self):
        if self.combustivel == "gasolina":
            return 20 * self.quilometragem
        if self.combustivel == "alcool":
            return 10 * self.quilometragem
        if self.combustivel == "hibrido":
            return 5 * self.quilometragem

File: Extra/Ruby/PegadaDeCarbono/carro.rb
require './pegada_de_carbono.rb'

class Carro
  include PegadaDeCarbono

  attr_accessor :marca, :placa, :quilometragem, :combustivel
  
  def initialize(marca, placa, quilometragem, combustivel)
    @marca = marca
    @placa = placa
    @quilometragem = quilometragem
    @combustivel = combustivel
  end

  def print
    puts "CARRO"
    puts "Marca: #{@marca}"
    puts "Placa: #{@placa}"
    puts "Quilometragem: #{@quilometragem}"
    puts "Combustivel: #{@combustivel}"
    puts "--------------------"
  end

  def move(distancia)
    @quilometragem += distancia
  end

  def getPegadaCarbono
    if @combustivel == "gasolina"
      return 20 * @quilometragem
    elsif @combustivel == "alcool"
      return 10 * @quilometragem
    elsif @combustivel == "hibrido"
      return 15 * @quilometragem
    end
  end
  
end
